ath79: fix calibration size for AR9285

neheb opened the pull request:
These devices use AR9285, which uses 3d8 as the calibration size, not 440 like newer chips do. Add a compatible line to make it clear that this is the case.

ping @DragonBluep

neheb committed:
ath79: fix calibration size for AR9285

These devices use AR9285, which uses 1f8 as the calibration size, not
440 like newer chips do. Actually the driver mandates a minimum of 200.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19863
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented:
@DragonBluep ugh based on

https://github.com/openwrt/openwrt/pull/17278#issuecomment-2544574689

it seems it should be 0x200 instead.

Djfe commented:
read again
AR9271&&AR9285 != AR9287

AR9287 is 0x3d8

neheb commented:
this device has

```
compatible = "pci168c,002b";
```

which is AR9285

according to Documentation/devicetree/bindings/net/wireless/qca,ath9k.yaml

Djfe commented:
thanks for clarification, feel free to correct your top post

> These devices use AR9287,

neheb committed:
ath79: add nvmem handling for AR9285 devices

They have the same 0x200 calibration size.

Added various compatible lines in various places to make it clear what
device we're talking about.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19863
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented:
I only did that as dtsi files were pointing to devices with different calibrations.
